File: uncompress/uncompress.js
// This doesn't work if the integer is more than one digit
const uncompress = (s) => {
  let split = s.split("");

  let nums = [];
  let letter = [];

  for (let i = 0; i < split.length; i++) {
    if (Number(split[i])) {
      nums.push(split[i]);
    } else {
      letter.push(split[i]);
    }
  }
  let result = [];
  for (let i = 0; i < nums.length; i++) {
    result.push(letter[i].repeat(nums[i]));
  }
  return result.join("");
};

File: Structy_1/02_Array_and_String/03_uncompress/uncompress.py
# My Solution
def uncompress(s):
  nums = '0123456789'
  i = 0
  j = 0
  result = []
  
  while j< len(s):
    if s[j] in nums:
      j += 1
    else:
      num = int(s[i:j])
      result.append(num * s[j])
      j += 1
      i = j
  return "".join(result)

print(uncompress("2c3a1t")) # -> 'ccaaat'

# Alvin's Solution
def uncompress(s):
  numbers = '0123456789'
  result = []
  i = 0
  j = 0
  while j < len(s):
    if s[j] in numbers:
      j += 1
    else:      
      num = int(s[i:j])
      result.append(s[j] * num)
      j += 1
      i = j
      
  return ''.join(result)